Doc/library/code.rst

```diff
@@ -65,6 +65,32 @@ build applications which provide an interactive interpreter prompt.
    raises :exc:`OverflowError` or :exc:`ValueError` if the command contains an
    invalid literal.
 
+Overriding Console Output
+-------------------------
+
+The output for :class:`InteractiveConsole` and :class:`InteractiveInterpreter`
+is written using one of :data:`sys.stderr`, :func:`sys.displayhook`
+or :func:`sys.excepthook`. Which output mechanism is used depends on the
+reason for producing output:
+
+* The return value from successfully interpreted Python statements are printed
+  by calling :func:`sys.displayhook`. This is a side effect of
+  :func:`compile_command` passing ``'single'`` for *symbol*.
+* If the default :func:`sys.excepthook` has not been replaced, syntax errors
+  and exception tracebacks are printed by calling the
+  :meth:`write <io.TextIOBase.write>` method on :data:`sys.stderr`.
+* If :func:`sys.excepthook` has been replaced with a user defined function
+  :func:`sys.excepthook` is called to handle printing of exception tracebacks
+  and syntax errors.
+
+Additionally, :class:`InteractiveConsole` will print banner information to
+:data:`sys.stderr` if *banner* was passed to the constructor.
+
+To customize the console output a user should replace :func:`sys.displayhook`
+and :func:`sys.excepthook` with custom implementations. Alternatively, users
+can replace the :meth:`InteractiveConsole.write` method on a subclass
+instead of replacing :func:`sys.excepthook` to control the printing of errors.
+
```

Comment:
I think this can be much shorter, it's a bit repetitive and some of the information is not really needed (it described what the code does rather than what the user needs to know).

Some examples:

"printed by calling the  :meth:`write <io.TextIOBase.write>` method on :data:`sys.stderr`"  is just "written to :data:`sys.stderr`".

The second and third bullets can be merged into one and made shorter:
"Exception tracebacks and syntax errors are written to :data:`sys.stderr` by default but can be redirected by setting :func:`sys.excepthook`".

The last paragraph starting with "To customize..." I would move it to be beginning and make it shorter (the first half repeats what was already said).